server/main.lua

```diff
@@ -78,38 +89,61 @@ function GiveKeys(id, plate)
     local Player = QBCore.Functions.GetPlayer(id)
     if not Player then return end
     local citizenid = Player.PlayerData.citizenid
+
     if not plate then
         if GetVehiclePedIsIn(GetPlayerPed(id), false) ~= 0 then
             plate = QBCore.Shared.Trim(GetVehicleNumberPlateText(GetVehiclePedIsIn(GetPlayerPed(id), false)))
         else
             return
         end
     end
+
     if not VehicleList[plate] then VehicleList[plate] = {} end
     VehicleList[plate][citizenid] = true
+
+    local oldKeys = Player.PlayerData.metadata["vehicleKeys"] or {}
+    oldKeys[plate] = true
+    Player.Functions.SetMetaData("vehicleKeys", oldKeys)
+
     TriggerClientEvent('QBCore:Notify', id, Lang:t('notify.vgetkeys'))
     TriggerClientEvent('qb-vehiclekeys:client:AddKeys', id, plate)
 end
 
 exports('GiveKeys', GiveKeys)
 
 function RemoveKeys(id, plate)
-    local citizenid = QBCore.Functions.GetPlayer(id).PlayerData.citizenid
+    local Player = QBCore.Functions.GetPlayer(id)
+    if not Player then return end
+    local citizenid = Player.PlayerData.citizenid
 
     if VehicleList[plate] and VehicleList[plate][citizenid] then
         VehicleList[plate][citizenid] = nil
     end
 
+    local oldKeys = Player.PlayerData.metadata["vehicleKeys"] or {}
+    oldKeys[plate] = nil
+    Player.Functions.SetMetaData("vehicleKeys", oldKeys)
+
     TriggerClientEvent('qb-vehiclekeys:client:RemoveKeys', id, plate)
 end
 
 exports('RemoveKeys', RemoveKeys)
 
 function HasKeys(id, plate)
-    local citizenid = QBCore.Functions.GetPlayer(id).PlayerData.citizenid
+    local Player = QBCore.Functions.GetPlayer(id)
+    if not Player then return false end
```

Comment:
```suggestion
    if not Player then
        return false
    end
```

better readability

Reply:
I think is better to keep it one line, it's cleaner from my view